obspy/core/utcdatetime.py

```diff
@@ -30,6 +30,12 @@
 # is called a lot. Thus pre-compile it.
 _YEAR0REGEX = re.compile(r"^(\d{1,3}[-/,])(.*)$")
 
+_ISO8601_REGEX = re.compile(
+    r"^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?"
+    r"|W([0-4]\d|5[0-3])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6"
+    r"])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17"
+    r"[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$")
```

Comment:
Wow.

Reply:
Yea, uh, maybe you want to use `re.VERBOSE` on this one? Or at least don't wrap in the middle of character sets.